tox.ini

```diff
@@ -0,0 +1,75 @@
+# This is a template `tox.ini` file for snaps
+# This file is managed by bootstack-charms-spec and should not be modified
+# within individual snap repos. https://launchpad.net/bootstack-charms-spec
+
+[tox]
+skipsdist=True
+envlist = lint, unit, func
+skip_missing_interpreters = True
+
+[testenv]
+basepython = python3
+allowlist_externals =
+    pytest
+
+[testenv:dev-environment]
+envdir = {toxinidir}/.venv
+deps =
+    pre-commit
+    {[testenv:lint]deps}
+    {[testenv:unit]deps}
+    {[testenv:func]deps}
+commands =
+    pre-commit install
+
+[testenv:pre-commit]
+envdir = {[testenv:dev-environment]envdir}
+deps = {[testenv:dev-environment]deps}  # ensure that dev-environment is installed
+commands = pre-commit run --all-files
+
+[testenv:lint]
+commands =
+    pflake8
+    pylint --recursive=y .
+    mypy . --install-types --non-interactive
+    black --check --diff --color .
+    isort --check --diff --color .
+deps =
+    black
+    flake8
+    pyproject-flake8
+    flake8-docstrings
+    pep8-naming
+    flake8-colors
+    colorama
+    isort
+    pylint
+    mypy
+    {[testenv:unit]deps}
+    {[testenv:func]deps}
+
+[testenv:reformat]
+envdir = {toxworkdir}/lint
+commands =
+    black .
+    isort .
+deps = {[testenv:lint]deps}
+
+[testenv:unit]
+commands = pytest {toxinidir}/tests/unit \
+    {posargs:-v --cov --cov-report=term-missing --cov-report=html --cov-report=xml}
+deps =
+    -r {toxinidir}/requirements.txt
+    -r {toxinidir}/tests/unit/requirements.txt
+    pytest
+    pytest-cov
+setenv = PYTHONPATH={toxinidir}
```

Comment:
This is not current version, can you check all templates again to be in current version.